src/client/api/courses.ts

```diff
@@ -55,6 +56,21 @@ export const getCourseScheduleForSemester = async (
   return response.data as ScheduleViewResponseDTO[];
 };
 
+/**
+ * Given a room id, calendar year, and term, this will retrieve the course
+ * information for the courses that will occur in the requested room and
+ * semester.
+ */
+export const getRoomScheduleForSemester = async (
+  roomId: string,
+  calendarYear: number,
+  term: TERM
+): Promise<RoomScheduleResponseDTO[]> => {
+  const response = await request
+    .get(`/api/course-instances/room-schedule?roomId=${roomId}&term=${term}&year=${calendarYear}`);
```

Comment:
Instead of using a template string, you can pass the query parameters in the `params` field of a [RequestConfig][1] object as the second argument to `get()`, i.e.

```ts
const response = awaite request
  .get(
    '/api/course-instances/room-schedule',
    {
      params: {
        roomId,
        term,
        year: calendarYear,
      }
    }
  )
```

Functionally it should be exactly the same, but the config object version would, in theory, better handle `undefined` values and uri encoding for special characters. Neither of those is really a concern with how this is currently used, but would be good to avoid in general

[1]: https://axios-http.com/docs/req_config